pkgs/desktops/gnome-3/3.20/core/libpeas/default.nix

```diff
@@ -1,13 +1,15 @@
 { stdenv, fetchurl, pkgconfig, intltool, gnome3
-, glib, gtk3, gobjectIntrospection, python, pygobject3
+, glib, gtk3, gobjectIntrospection, python3, python3Packages, ncurses, python2, python2Packages
 }:
 
 stdenv.mkDerivation rec {
   inherit (import ./src.nix fetchurl) name src;
 
+  configureFlags = [ "--enable-python3" "--enable-python2" ];
+
   buildInputs =  [
-   intltool pkgconfig glib gtk3 gobjectIntrospection python pygobject3
-   gnome3.defaultIconTheme
+   intltool pkgconfig glib gtk3 gobjectIntrospection python3 python3Packages.pygobject3 python2
+   gnome3.defaultIconTheme ncurses
```

Comment:
I would not expect that. 

From the manual:
```
--references
Prints the set of references of the store paths paths, that is, their immediate dependencies. (For all dependencies, use --requisites.)
```
We haven't explicitly stated it.

Now what can happen is that during building it somehow managed to get a link to the interpreter, and stored that. The path of the store object corresponding to that path is then added to the references, if I am understanding this correctly. But why does it behave differently with Python 2 vs. Python 3?

Building the following file
```nix
with import <nixpkgs> {};
with pkgs;

let
  python2 = python2Packages.python.withPackages ( ps: with ps; [ pygobject3 ] );
  python3 = python3Packages.python.withPackages ( ps: with ps; [ pygobject3 ] );
in stdenv.mkDerivation {
  name = "doublepython";

  unpackPhase = "true";
  installPhase = ''
    mkdir -p $out
    ln -s -t $out ${python2} 
    ln -s -t $out ${python3} 
  '';
  buildInputs =  [ intltool pkgconfig glib gtk3 gnome3.defaultIconTheme ncurses ];

  propagatedBuildInputs = [ python2 python3 gobjectIntrospection ];
}
```
doesn't show any extra Python interpreters, implying that there is at least nothing wrong with our inputs/dependencies.

Therefore I think it has to be something in `libpeas` that manages to get the path to the actual interpreter and stores it somewhere.

And indeed, searching for the 3.5.1 interpreter directly gives a hit
```
/nix/store/7hdcc7c9jqg5fjaj7lp4b87sql98apl6-libpeas-1.18.0$ grep -r "python3-3.5.1" .
./nix-support/propagated-native-build-inputs:/nix/store/5wyc68zp1qj05ac5xgp18v8qpf259m96-python-2.7.11-env /nix/store/2vxw8wqhkwavr9fj825ihgkk01mv3j13-python3-3.5.1-env /nix/store/3wjwcx3918dr9kwfiabxhrlhj3hbjq6f-gobject-introspection-1.48.0-dev
Binary file ./lib/libpeas-1.0/loaders/libpython3loader.so matches
./lib/libpeas-1.0/loaders/libpython3loader.la:dependency_libs=' -L/nix/store/w9gwivwr7ih57b4y1dfhv64xy316p5qn-python3-3.5.1/lib /nix/store/7hdcc7c9jqg5fjaj7lp4b87sql98apl6-libpeas-1.18.0/lib/libpeas-1.0.la -L/nix/store/nki5m34crppfvzcm71my04vhh3z8x76k-glib-2.48.1/lib -L/nix/store/nym478rhdc2ljk1pqfgic8cs7qh6a2wy-gobject-introspection-1.48.0/lib -L/nix/store/w19fpfdx3kf6q19advyp4i9xl5w8h8pv-pcre-8.38/lib -L/nix/store/q7nkzs3rdj6f3ryksknmdn1wpflivmmb-libffi-3.2.1/lib/../lib64 -L/nix/store/sx0v855xbxq1vjx79dprqrfjqzx9dbcz-zlib-1.2.8/lib /nix/store/nym478rhdc2ljk1pqfgic8cs7qh6a2wy-gobject-introspection-1.48.0/lib/libgirepository-1.0.la /nix/store/nki5m34crppfvzcm71my04vhh3z8x76k-glib-2.48.1/lib/libgio-2.0.la /nix/store/nki5m34crppfvzcm71my04vhh3z8x76k-glib-2.48.1/lib/libgmodule-2.0.la -lz -lresolv /nix/store/nki5m34crppfvzcm71my04vhh3z8x76k-glib-2.48.1/lib/libgobject-2.0.la /nix/store/q7nkzs3rdj6f3ryksknmdn1wpflivmmb-libffi-3.2.1/lib/../lib64/libffi.la /nix/store/nki5m34crppfvzcm71my04vhh3z8x76k-glib-2.48.1/lib/libglib-2.0.la /nix/store/w19fpfdx3kf6q19advyp4i9xl5w8h8pv-pcre-8.38/lib/libpcre.la -lpython3.5m -lpthread -ldl -lcrypt -lncurses -lutil -lm'
```
whereas grepping for 2.7.11 doesn't.
```
/nix/store/7hdcc7c9jqg5fjaj7lp4b87sql98apl6-libpeas-1.18.0$ grep -r "python-2.7.11" .
./nix-support/propagated-native-build-inputs:/nix/store/5wyc68zp1qj05ac5xgp18v8qpf259m96-python-2.7.11-env /nix/store/2vxw8wqhkwavr9fj825ihgkk01mv3j13-python3-3.5.1-env /nix/store/3wjwcx3918dr9kwfiabxhrlhj3hbjq6f-gobject-introspection-1.48.0-dev
Binary file ./lib/libpeas-1.0/loaders/libpythonloader.so matches
```